### ISSUE TITLE
redundant-fn-wrapper linter

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#1526]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

`(fn [x] (foo x))` or `#(foo %)` rather than `foo` should give a warning

A more complex case would be `(update x :foo #(bar % a b))` vs `(update x :foo bar a b)`, not sure if we should cover that too.

TODO:

- [x] port back the macroexpand of fn literal to master first
- [x] reverse detecting of redundant fn wrapper by looking in a function call upwards and then add info to call.
- [x] then in linters.clj check info for redundant-fn-wrapper and emit warning
- [x] false positives when wrapping macros. Limit linting only to known clojure vars that are not macros for now?
- [x] false positives when wrapping Java interop
- [x] false positive:
``` clojure
(defmethod route-view :default [match]
  [:div "unsupported route " (pr-str match)])
```
- [x] address false positives from script/diff
- [x] false positive `#(clojure.lang.RT/iter %)`
- [x] false positive `(fn [x] (inc x) (dec x))` 
- [x] write tests
